core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2858,8 +2858,8 @@ object ZIO extends ZIOCompanionPlatformSpecific {
           .forkManaged
         _ <- interrupter.use_ {
               ZIO
-                .whenM(result.await.map(!_)) {
-                  causes.get.flatMap(ZIO.halt(_))
+                .whenM(ZIO.foreach(fibers)(_.await).map(_.exists(!_.succeeded))) {
```

Comment:
Nothing is awaiting the `result` with this change. Is it still required?

Reply:
You're right. Deleted the old `result` and renamed `failureTrigger` to `result`.